### PR TITLE
WIP chore(website): docgen with typescript

### PR DESCRIPTION
**Pull request by cnishina**
This works! Except it appears that I am also creating a `website/website/docgen/build/toc.json`. @juliemr I need to know more dgeni to fix this.

### CODE REVIEW

**Comment by juliemr** [comment]
It's OK googlebot :)

**Comment by cnishina** [comment]
Hmmm... worked on my mac. Now it doesn't work on my desktop. :confused:

File: website/docgen/dgeni-config.js [hunk 1]
```diff
@@ -57,29 +60,31 @@ myPackage.factory(require('./inline_tags/code'))
...
-    {include: 'lib/selenium-webdriver/webdriver.js'},
-    {include: 'lib/webdriver-js-extender/index.js'}
-  ];
+  // Go to the protractor project root.
+  // --- JS
+  readFilesProcessor.basePath = path.resolve(__dirname, '..');
+  readFilesProcessor.sourceFiles = [{
+      include: 'lib/**/*.js',
+      exclude: 'lib/**/node_modules/**/*',
+      basePath: 'lib'
```
Comment (juliemr) [inline website/docgen/dgeni-config.js]: Why do we need `basePath` both here and on `readFilesProcessor.basePath`?

File: website/docgen/dgeni-config.js [hunk 2]
```diff
@@ -57,29 +60,31 @@ myPackage.factory(require('./inline_tags/code'))
...
-    {include: 'built/expectedConditions.js'},
-    {include: 'lib/selenium-webdriver/locators.js'},
-    {include: 'lib/selenium-webdriver/webdriver.js'},
-    {include: 'lib/webdriver-js-extender/index.js'}
-  ];
+  // Go to the protractor project root.
+  // --- JS
+  readFilesProcessor.basePath = path.resolve(__dirname, '..');
+  readFilesProcessor.sourceFiles = [{
+      include: 'lib/**/*.js',
```
Comment (juliemr) [inline website/docgen/dgeni-config.js]: I think this `*.js` is probably dangerous - we only want to get the specific files we listed before. I guess just the `selenium-webdriver` ones in this case.

**Comment by cnishina** [comment]
Just kidding. I had some cached files that made it looked like it worked. This does not work.

**Pull request closed without merging**